tests/test_project/tests/test_admin.py

```diff
@@ -111,6 +112,12 @@ def test_context_processor(self):
         response = self.client.get(url)
         self.assertContains(response, 'class="shelf"')
 
+    def test_can_see_unpermitted_projects(self):
+        Permission.objects.filter(codename__endswith='shelf').delete()
```

Comment:
Ok. Even though i thought we are focusing on the fact that if a user have a `permission` on a module he should see its menu item, otherwise he doesn't see it irrespective of who he is.